tidy_data/tidy_data.md

```diff
@@ -0,0 +1,737 @@
+
+</div>
+
+### Column Headers as Values
+
+Let's consider an example from section 3.1 of [Hadley's article](https://www.jstatsoft.org/index.php/jss/article/view/v059i10/772).  The column headers in this table, taken from Pew Forum, contain values -- income values.  This table is readable for humans, and takes advantage of horizontal space on a page.  It makes sense that it would appear in a report for human consumption.  This data is messy, however, because it violates one of the important rules about tidy data.
+
+<div class = "hint"> Note that in the world of R users, Hadley Wickham is considered such a singularly important and yet friendly figure that he is often referred to simply as "Hadley".
+</div>
+
+| religion | <lia-keep><$10k</lia-keep> | <lia-keep>$10–20k</lia-keep> | <lia-keep>$20–30k</lia-keep> | <lia-keep>$30–40k</lia-keep> | <lia-keep>$40–50k</lia-keep> | <lia-keep>$50–75k</lia-keep> |
+| -------- | ----- | ------- | ------- | ------- | ------- | ------- |
+| Agnostic | 27 | 34 | 60 | 81 | 76 | 137 |
+| Atheist | 12 | 27 | 37 | 52 | 35 | 70 |
+| Buddhist | 27 | 21 | 30 | 34 | 33 | 58 |
+| Catholic | 418 | 617 | 732 | 670 | 638 | 1116 |
+| Don't know/refused | 15 | 14 | 15 | 11 | 10 | 35 |
+| Evangelical Prot | 575 | 869 | 1064 | 982 | 881 | 1486 |
+| Hindu | 1 | 9 | 7 | 9 | 11 | 34 |
+| Historically Black Prot | 228 | 244 | 236 | 238 | 197 | 223 |
+| Jehovah's Witness | 20 | 27 | 24 | 24 | 21 | 30 |
+| Jewish | 19 | 19 | 25 | 25 | 30 | 95 |
+
+
+While humans can make sense of this data, remember that computers must be told in sometimes agonizing detail what to do.  If we were to ask "which is the largest group?" using the table above, we'd have to say (in code, of course) something along the lines of "look inside each row, at columns 2-7, and take the largest value of all of those cells, and then take the value of the first column of the row with that largest-valued cell, and the column header of that cell, and that's your group".
+
+<div class = "hint">
+
+<div style = "align-items: center; display: flex;">
+<div style = "margin: 1rem; max-width: 55%; float:left; padding-right:4em;">
+
+Thinking in computer-ese, giving very detailed instruction, is a kind of **algorithmic thinking**.  An **algorithm** is simply a well-specified process that solves a problem.  It takes practice to think like a computer instead of like a person, and one of the fun things that many programmers like to do to help this process along is called "rubber duck debugging".  This involves explaining to, yes, a rubber duck (or other friendly yet not very smart collaborator, like your cat) exactly what you're trying to do and how to get there, following detailed rules.  It may seem silly, but give it a try!
+
+</div>
+<div style = "margin: 1rem auto; max-width: 35%; float:left;">
+![Rubber duck](media/rubber_duck.png)
```

Comment:
Alt text reads more naturally with a screenreader when it ends with punctuation like a period.

```suggestion
![Rubber duck.](media/rubber_duck.png)
```

Reply:
Agree, changed